src/Agent.Sdk/ProcessInvoker.MacLinux.cs

```diff
@@ -13,21 +13,27 @@ public sealed partial class ProcessInvoker : IDisposable
     {
         private async Task<bool> SendPosixSignal(PosixSignals signal, TimeSpan timeout)
         {
+            if (_proc == null)
+            {
+                Trace.Info($"Process already exited, no need to send {signal}.");
```

Comment:
Is signal going to serialize correctly here?

Reply:
it was serialized like that below, so I assume so